tarogarden/planter.go

```diff
@@ -486,6 +491,19 @@ func (c *ChainPlanter) prepTaroSeedling(ctx context.Context,
 		defer cancel()
 		err = c.cfg.Log.CommitMintingBatch(ctx, newBatch)
 		if err != nil {
+			// Handle SQLite asset seedling name unique constraint
+			// violation error.
+			var sqlError *sqlite.Error
```

Comment:
Cool this is looking correct now, final comment here is to actually push this further into the `CommitMintingBatch` implementation. This way we can hide the fact that this is backed by a SQL DB from the planter itself (we're also leaking things like the table/column name here as well, tho we don't need to). 

So we'd move these checks to the _end_ of the current `CommitMintingBatch` implementation (catch the error, then return `ErrDuplicateSeedlingName`).

Reply:
That makes sense. I can handle the postgres error at that point also.

And then I'll also move the `ErrDuplicateSeedlingName` definition to `tarodb` to avoid a circular dependency right?

Or a general db error and then another `tarogardner` error?

Reply:
I think we can actually have `ErrDuplicateSeedlingName` here w/o creating a circular dep? Since already `tarodb` returns types defined in this package.

Reply:
`ErrDuplicateSeedlingName` definition remains in `tarogarden`, no circular dep problem. I thought there might be on the package level.